verilog/tools/kythe/indexing_facts_tree_extractor_test.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright 2017-2020 The Verible Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "indexing_facts_tree_extractor.h"
+#include "common/text/concrete_syntax_tree.h"
+#include "gtest/gtest.h"
+#include "verilog/analysis/verilog_analyzer.h"
+
+#undef EXPECT_OK
+#define EXPECT_OK(value) EXPECT_TRUE((value).ok())
+
+namespace verilog {
+namespace kythe {
+namespace {
+
+typedef IndexingFactNode T;
+
+bool FactsTreeEqual(const IndexingNodeData& lhs, const IndexingNodeData& rhs) {
+  auto left = lhs, right = rhs;
+  bool equal = left.GetIndexingFactType() == right.GetIndexingFactType();
+
+  const auto &lAnchors = left.Anchors(), rAnchors = right.Anchors();
+  equal &= lAnchors.size() == rAnchors.size();
+  for (size_t i = 0; i < lAnchors.size(); i++) {
+    equal &= lAnchors[i].StartLocation() == rAnchors[i].StartLocation() &&
+             lAnchors[i].EndLocation() == rAnchors[i].EndLocation() &&
+             lAnchors[i].Value() == rAnchors[i].Value();
+  }
+
+  return equal;
+}
```

Comment:
Add a test for `FactsTreeEqual` so that we know it properly returns `false` when there is actually a difference.
(Otherwise, the above function could be replaced with`return true;` and tests would still pass!)

Also recommend comparator tests for `Anchor::operator==` after you add it.